### PR TITLE
[DEM]fix addvariables by default

### PR DESCRIPTION
**Pull request by farrufat-cimne**
Currently being added to spheres, rigidface and clusters.
Added  flag pointing to json.
Is it actually required to be added in clusters variables too?

### CODE REVIEW

**Review by maceligueta — DISMISSED** [review]

File: applications/DEMApplication/python_scripts/DEM_procedures.py [hunk 1]
```diff
@@ -1530,9 +1533,10 @@ def AddGlobalVariables(self):
             self.PushPrintVar(self.PostAngularVelocity, ANGULAR_VELOCITY, self.global_variables)
         if self.DEM_parameters["PostParticleMoment"].GetBool():
             self.PushPrintVar(self.PostParticleMoment, PARTICLE_MOMENT, self.global_variables)
-        self.PushPrintVar(self.PostControlModule, TARGET_STRESS, self.global_variables)
-        self.PushPrintVar(self.PostControlModule, REACTION_STRESS, self.global_variables)
-        self.PushPrintVar(self.PostControlModule, LOADING_VELOCITY, self.global_variables)
+        if self.DEM_parameters["PostControlModule"].GetBool():
+            self.PushPrintVar(self.PostControlModule, TARGET_STRESS, self.global_variables)
+            self.PushPrintVar(self.PostControlModule, REACTION_STRESS, self.global_variables)
+            self.PushPrintVar(self.PostControlModule, LOADING_VELOCITY, self.global_variables)
```
Comment (ipouplana) [inline applications/DEMApplication/python_scripts/DEM_procedures.py]: I could be wrong, but I think that the 'self.PostControlModule' is a bool used to check whether the variable should be printed or not, and so the if statement in line 1536 could be not necessary.

File: applications/DEMApplication/python_scripts/DEM_procedures.py [hunk 2]
```diff
@@ -608,9 +610,10 @@ def AddClusterVariables(self, model_part, DEM_parameters):
             model_part.AddNodalSolutionStepVariable(EULER_ANGLES)
 
         # CONTROL MODULE
-        model_part.AddNodalSolutionStepVariable(TARGET_STRESS)
-        model_part.AddNodalSolutionStepVariable(REACTION_STRESS)
-        model_part.AddNodalSolutionStepVariable(LOADING_VELOCITY)
+        if DEM_parameters["PostControlModule"].GetBool():
+            model_part.AddNodalSolutionStepVariable(TARGET_STRESS)
+            model_part.AddNodalSolutionStepVariable(REACTION_STRESS)
+            model_part.AddNodalSolutionStepVariable(LOADING_VELOCITY)
```
Comment (ipouplana) [inline applications/DEMApplication/python_scripts/DEM_procedures.py:616]: I agree that control module variables should only be added when somebody uses the control module, but we need to think of a cleaner way to do it. With that if statement, the usage of control module depends on whether you print the control module variables or not, which is not the most appropriate (you could perfectly be using the control module without printing those variables).

**Review by ipouplana — COMMENTED** [review]
I am not blocking this PR, but I believe that we need to think which is the best way to manage the variables related to control module process.

And regarding the need of adding control module variables to clusters... Couldn't there be cases where we have a continuum DEM body made of clusters of spheres over which a control module is applied ?

File: applications/DEMApplication/python_scripts/DEM_procedures.py (hunk 1 above)
Comment (farrufat-cimne) [reply applications/DEMApplication/python_scripts/DEM_procedures.py]: @ipouplana

Comment (farrufat-cimne) [reply applications/DEMApplication/python_scripts/DEM_procedures.py]: fixed

**Review by ipouplana — APPROVED** [review]

**Pull request merged**